utilities/cluster.py

```
@@ -17,3 +20,41 @@ def cache_admin_client() -> DynamicClient:
+            timeout=sleep,
+        )
+        return stdout.strip() if did_succeed else ""
+
+    for result in TimeoutSampler(
+        wait_timeout=wait_timeout,
+        sleep=sleep,
+        func=_whoami,
+        exceptions_dict={TimeoutExpired: []},
+    ):
+        if result:
+            return result
```

Ruff (0.15.10)

[warning] Missing explicit `return` at the end of function able to return non-`None` value

Add explicit `return` statement

(RET503)